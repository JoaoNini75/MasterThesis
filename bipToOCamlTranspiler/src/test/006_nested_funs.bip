let test1_outer () : int = begin
  let x = 10 in

  let test1_inner (i : int) : int = begin
    x * i 
  end
  (*@ ensures true *)
	in

  let i = 2 in
  test1_inner (i)
end
(*@ ensures true *) 

let funs_nested (x) = begin

  let fun_inner (i) = begin
		let i = ref 0 in
		while !i < 3 do
			i := !i + 2;
			i := !i - 1
		done;
		!i
  end
	(*@ ensures true *)
	in

  fun_inner (x)
end
(*@ ensures true *) 
