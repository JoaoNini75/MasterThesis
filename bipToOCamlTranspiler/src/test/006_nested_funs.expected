let test1_outer () : int =
  let x = 10 in

  let test1_inner (i : int) : int =
    (x * i)
  (*@ ensures true *)
  in
    
  let i = 2 in
  test1_inner (i)
(*@ ensures true *)

let funs_nested (x) =

  let fun_inner (i) =
    let i = ref (0) in

    while (!i < 3) do
      i := (!i + 2);
      i := (!i - 1)
    done;

    !i
  (*@ ensures true *)
  in
    
  fun_inner (x)
(*@ ensures true *)


